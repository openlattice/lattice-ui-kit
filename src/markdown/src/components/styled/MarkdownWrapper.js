import styled from 'styled-components';

const MarkdownWrapper = styled.div`
  code {
    padding: .2em .4em;
    margin: 0;
    font-size: 85%;
    background-color: rgba(27,31,35,.05);
    border-radius: 3px;
  }

  .hljs{
    display: block;
    overflow-x: auto;
    padding: .5em;
    color: #333;
    background: #f8f8f8;
  }

  .hljs-comment, .hljs-quote {
    color: #998;
    font-style: italic;
  }

  .hljs-keyword, .hljs-selector-tag, .hljs-subst {
    color: #333;
    font-weight: 700
  }

  .hljs-literal, .hljs-number, .hljs-tag .hljs-attr, .hljs-template-variable, .hljs-variable{
    color: teal;
  }

  .hljs-doctag, .hljs-string {
    color: #d14;
  }
  .hljs-section, .hljs-selector-id, .hljs-title {
    color: #900;
    font-weight: 700;
  }
  .hljs-subst {
    font-weight: 400;
  }
  .hljs-class .hljs-title, .hljs-type {
    color: #458;
    font-weight: 700;
  }

  .hljs-attribute, .hljs-name, .hljs-tag {
    color: navy;
    font-weight: 400;
  }

  .hljs-link, .hljs-regexp {
    color: #009926;
  }

  .hljs-bullet, .hljs-symbol {
    color: #990073;
  }

  .hljs-built_in, .hljs-builtin-name {
    color: #0086b3;
  }

  .hljs-meta{
    color: #999;
    font-weight: 700;
  }

  .hljs-deletion {
    background: #fdd;
  }

  .hljs-addition{
    background: #dfd;
  }

  .hljs-emphasis {
    font-style: italic;
  }

  .hljs-strong{
    font-weight: 700;
  }

  table {
    display: block;
    width: 100%;
    overflow: auto;
    border-spacing: 0;
    border-collapse: collapse;
  }

  table th {
    font-weight: 600;
  }

  table td,
  table th {
    padding: 6px 13px;
    border: 1px solid #dfe2e5;
  }

  table tr {
    background-color: #fff;
    border-top: 1px solid #c6cbd1;
  }

  table tr:nth-child(2n) {
    background-color: #f6f8fa;
  }

`;

export {
  MarkdownWrapper
};
