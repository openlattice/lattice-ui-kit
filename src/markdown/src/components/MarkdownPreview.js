// @flow

import type { Node } from 'react';

import ReactMarkdown from 'react-markdown';
import rehypeHighlight from 'rehype-highlight';
import remarkGFM from 'remark-gfm';

import MarkdownWrapper from './styled/MarkdownWrapper';

type Props = {
  children :Node;
};

const MarkdownPreview = ({ children } :Props) => (
  <MarkdownWrapper>
    <ReactMarkdown
        rehypePlugins={[rehypeHighlight]}
        remarkPlugins={[remarkGFM]}>
      {children}
    </ReactMarkdown>
  </MarkdownWrapper>
);

export default MarkdownPreview;
