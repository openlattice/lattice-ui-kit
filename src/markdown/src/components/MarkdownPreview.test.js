import ReactMarkdown from 'react-markdown';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import MarkdownPreview from './MarkdownPreview';
import MarkdownWrapper from './styled/MarkdownWrapper';

describe('MarkdownPreview', () => {

  test('render matches snapshot', () => {
    const wrapper = shallow(<MarkdownPreview />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should render MarkdownWrapper', () => {
    const wrapper = shallow(<MarkdownPreview />);

    expect(wrapper.find(MarkdownWrapper)).toHaveLength(1);
  });

  test('should render ReactMarkdown', () => {
    const wrapper = shallow(<MarkdownPreview />);

    expect(wrapper.find(ReactMarkdown)).toHaveLength(1);
  });

});
