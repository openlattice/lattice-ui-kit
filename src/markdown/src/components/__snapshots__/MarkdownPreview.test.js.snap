// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownPreview render matches snapshot 1`] = `
<styled.div>
  <ReactMarkdown
    rehypePlugins={
      Array [
        Array [
          [Function],
          Object {
            "ignoreMissing": true,
          },
        ],
      ]
    }
    remarkPlugins={
      Array [
        [Function],
      ]
    }
    transformLinkUri={[Function]}
  />
</styled.div>
`;
